### ISSUE TITLE
protabled: don't insist on ".raw" suffix for raw images so strictly

### ISSUE THREAD

**Issue by ruuda** [systemd/systemd#10095]
**systemd version the issue has been seen with**
> 239

**Used distribution**
 > Arch

**Expected behaviour you didn't see**
 > I have a squashfs image `image.img`. If I mount that at `/tmp/image`, then `sudo /lib/systemd/portablectl inspect /tmp/image` reports metadata fine. I expected `sudo /lib/systemd/portablectl inspect image.img` to work as well.

**Unexpected behaviour you saw**
 > The command printed `Failed to inspect image metadata: Wrong medium type`. `portablectl attach` also prints the `Wrong medium type` message.

**Steps to reproduce the problem**
<details><summary>First build a squashfs image that reproduces the problem.</summary>

Save the following to `repro.nix`.
```nix
with (import (fetchTarball {
  url = "https://github.com/NixOS/nixpkgs/archive/56b9f6fc8e1c3a4ad10ff7c61e461d7b7e038833.tar.gz";
  sha256 = "0v5y4wjfxbappsaibw88h7n1flcx7kpvg51mjv3h9m4aa3fn2c8q";
}) {});

let
  imageDir = stdenv.mkDerivation {
    name = "repro-filesystem";
    buildCommand = ''
      mkdir -p $out/usr/lib
      mkdir -p $out/usr/lib/systemd/system
      echo "PORTABLE_PRETTY_NAME=repro" > $out/usr/lib/os-release
      touch $out/usr/lib/systemd/system/repro.service
    '';
  };
in
  stdenv.mkDerivation {
    name = "repro.img";
    nativeBuildInputs = [ squashfsTools ];
    buildInputs = [ imageDir ];
    buildCommand =
      ''
        mksquashfs ${imageDir} $out \
          -no-fragments      \
          -processors 1      \
          -all-root          \
          -b 1048576         \
          -comp xz           \
          -Xdict-size 100%   \
      '';
  }
```
Run [`nix build -f repro.nix --out-link repro.img`](https://nixos.org/nix/).
</details><br>

Then:
```
$ file $(realpath repro.img)
/nix/store/cildzy70a6jgk5cq1d3mnbm7bg2q2wxh-repro.img: Squashfs filesystem, little endian, version 4.0, 493 bytes, 7 inodes, blocksize: 1048576 bytes, created: Thu Jan  1 00:00:01 1970
$ mkdir /tmp/repro
$ sudo mount repro.img /tmp/repro
$ sudo /usr/lib/systemd/portablectl inspect /tmp/repro  
(Matching unit files with prefix 'repro'.)
Image:
        /tmp/repro
Portable Service:
        repro
Operating System:
        n/a
Unit files:
        repro.service
$ sudo /usr/lib/systemd/portablectl inspect ./repro.img
(Matching unit files with prefix 'repro.img'.)
Failed to inspect image metadata: Wrong medium type
```

**Comment by poettering** [comment]
hmm, interesting... would it be OK if you could give me access to an example image that triggers this? (no need to put that up publicly, would be sufficient to pass that to me personally

**Comment by ruuda** [comment]
Sure, here is the image produced from the steps above: [repro.img.gz](https://github.com/systemd/systemd/files/2411988/repro.img.gz) (gzipped because Github does not allow uploading otherwise).

**Comment by poettering** [comment]
Ah, this works if you rename it to ".raw". the logic insists on that suffix, and removes it when deriving the service names to copy out. it's pretty strict on that, in order to avoid ambiguities.

This is quite a pitfall I mus admit though. We should do something about this, either make this more permissive, or produce better error strings and document this.

**Comment by poettering** [comment]
Fix waiting in #10307
